### PR TITLE
Create LoyaltyCard

### PR DESCRIPTION
**Pull request by nkrishang**

### CODE REVIEW

File: contracts/DynamicNFT.sol [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+    function initialize(
+        address _defaultAdmin,
+        string memory _name,
+        string memory _symbol,
+        string memory _contractURI,
+        address[] memory _trustedForwarders,
+        address _saleRecipient,
+        address _royaltyRecipient,
+        uint128 _royaltyBps,
+        uint128 _platformFeeBps,
```
Comment (jakeloo) [inline contracts/DynamicNFT.sol:75]: flat platform fee? since i think that's the indsutry standard these days

File: contracts/DynamicNFT.sol [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+    }
+
+    /// @dev Burns `tokenId`. See {ERC721-_burn}.
+    function burn(uint256 tokenId) external virtual {
+        // note: ERC721AUpgradeable's `_burn(uint256,bool)` internally checks for token approvals.
+        _burn(tokenId, true);
+    }
+
+    /// @dev Burns `tokenId`. See {ERC721-_burn}.
+    function burnAsAdmin(uint256 tokenId) external virtual onlyRole(METADATA_ROLE) {
```
Comment (jakeloo) [inline contracts/DynamicNFT.sol:161]: wrong role

File: contracts/DynamicNFT.sol [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+        if (_pricePerToken == 0) {
+            return;
+        }
+
+        (address platformFeeRecipient, uint16 platformFeeBps) = getPlatformFeeInfo();
+
+        address saleRecipient = _primarySaleRecipient == address(0) ? primarySaleRecipient() : _primarySaleRecipient;
+
+        uint256 totalPrice = _quantityToClaim * _pricePerToken;
+        uint256 platformFees = (totalPrice * platformFeeBps) / MAX_BPS;
```
Comment (jakeloo) [inline contracts/DynamicNFT.sol:245]: flat fee? since that's the more recent use case

File: contracts/LoyaltyCard.sol [hunk 4]
```diff
@@ -0,0 +1,355 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.11;
+
```
Comment (joaquim-verges) [inline contracts/LoyaltyCard.sol:3]: where is the thirdweb ascii ?

File: contracts/LoyaltyCard.sol [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+import "./extension/Royalty.sol";
+import "./extension/PrimarySale.sol";
+import "./extension/PlatformFee.sol";
+import "./extension/Multicall.sol";
+import "./extension/PermissionsEnumerable.sol";
+import "./extension/DefaultOperatorFiltererUpgradeable.sol";
+import "./openzeppelin-presets/metatx/ERC2771ContextUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+
+contract LoyaltyCard is
```
Comment (joaquim-verges) [inline contracts/LoyaltyCard.sol:27]: would be great to have some doc here about what this contract is and what the main features are

**Comment by nkrishang** [comment]
The `LoyaltyCard` contract has been audited. The contract + audit fixes are now being worked on in this PR: https://github.com/thirdweb-dev/contracts/pull/414

**Pull request closed without merging**